gwcs/coordinate_frames.py

```diff
@@ -501,7 +504,12 @@ def _world_axis_object_classes(self):
 
     @property
     def _world_axis_object_components(self):
-        return [('temporal', 0, 'value')]
+        if isinstance(self.reference_frame.value, np.ndarray):
+            return [('temporal', 0, 'value')]
+
+        def offset_from_time_and_reference(time):
+            return (time - self.reference_frame).sec
+        return [('temporal', 0, offset_from_time_and_reference)]
```

Comment:
I don't understand this change. APE 14 (just checked it because I keep forgetting what these properties are for) states the third item in the tuple should be a method on the corresponding class in `_world_axis_object_classes`.
How is `_world_axis_object_components` used?

Reply:
To address this issue with Time in core, we extended the base API to allow arbitrary callables in both [components](https://github.com/astropy/astropy/blob/master/astropy/wcs/wcsapi/low_level_api.py#L156-L159) and [classes](https://github.com/astropy/astropy/blob/master/astropy/wcs/wcsapi/low_level_api.py#L190-L193) which works around limitations in the constructors of various classes.

This implementation follows the one in core for FITS WCS pretty closely: https://github.com/astropy/astropy/blob/master/astropy/wcs/wcsapi/fitswcs.py#L633-L642

Reply:
Is there an easy way to add a test which uses these `_world_axis_component_c*` methods?